SingularityService/src/main/java/com/hubspot/singularity/data/TaskManager.java

```diff
@@ -411,11 +411,19 @@ public int getNumHealthchecks(SingularityTaskId taskId) {
     return Maps.uniqueIndex(healthcheckResults, SingularityTaskIdHolder.getTaskIdFunction());
   }
 
-  @Timed
   public SingularityCreateResult saveTaskHistoryUpdate(SingularityTaskHistoryUpdate taskHistoryUpdate) {
+    return saveTaskHistoryUpdate(taskHistoryUpdate, false);
+  }
+
+  @Timed
+  public SingularityCreateResult saveTaskHistoryUpdate(SingularityTaskHistoryUpdate taskHistoryUpdate, boolean overwrite) {
     singularityEventListener.taskHistoryUpdateEvent(taskHistoryUpdate);
 
-    return create(getUpdatePath(taskHistoryUpdate.getTaskId(), taskHistoryUpdate.getTaskState()), taskHistoryUpdate, taskHistoryUpdateTranscoder);
+    if (overwrite) {
```

Comment:
makes sense -- can we further clarify the naming to something like `overwriteExisting`?